server/src/main/java/io/crate/planner/optimizer/symbol/rule/MoveSubscriptOnReferenceCastToLiteralCastInsideOperators.java

```diff
@@ -50,7 +52,7 @@ public MoveSubscriptOnReferenceCastToLiteralCastInsideOperators() {
             .with(f -> COMPARISON_OPERATORS.contains(f.name()))
             .with(f -> f.arguments().get(1).symbolType().isValueOrParameterSymbol())
             .with(f -> f.arguments().get(0), typeOf(Function.class).capturedAs(castCapture)
-                .with(f -> f.isCast())
+                .with(f -> f.name().equals(ImplicitCastFunction.NAME) || f.name().equals(TryCastFunction.NAME))
```

Comment:
shouldn't we have only the implicit cast here? the try_cast is also an explicit function, right?

Reply:
right, I used a wrong blueprint https://github.com/crate/crate/pull/12155/files.